### ISSUE TITLE
Does not work correctly on EC2 instance with IAM Instance Profile

### ISSUE THREAD

**Issue by vrivellino** [pgherveou/gulp-awspublish#45]
We have a project that uses gulp-awspublish to publish to S3, and we're trying to configure it on a build server that runs on an EC2 instance with an IAM Instance Profile assigned to it.

However, when executing our `deploy` task (which performs the upload), we getting a `Bad or invalid credentials` error:

```
...
[23:18:20] Starting 'dist:clean'...
[23:18:20] Finished 'dist:clean' after 3.51 ms
[23:18:20] Starting 'dist:copy'...
[23:18:20] Finished 'dist:copy' after 28 ms
[23:18:20] Starting 'dist:urls'...
[23:18:20] Finished 'dist:urls' after 2.77 ms
[23:18:20] Finished 'dist' after 9.45 s
[23:18:20] Starting 'deploy:clean'...
[23:18:21] Finished 'deploy:clean' after 655 ms
[23:18:21] Starting 'deploy:nogz'...
[23:18:21] 'deploy:nogz' errored after 666 μs
[23:18:21] Error in plugin 'gulp-awspublish'
Bad or invalid credentials
[23:18:21] 'deploy' errored after 10 s
[23:18:21] Error: [object Object]
    at formatError (/usr/lib/node_modules/gulp/bin/gulp.js:169:10)
    at Gulp.<anonymous> (/usr/lib/node_modules/gulp/bin/gulp.js:195:15)
    at Gulp.emit (events.js:117:20)
    at Gulp.Orchestrator._emitTaskDone (/home/ec2-user/dashboard/node_modules/gulp/node_modules/orchestrator/index.js:264:8)
    at /home/ec2-user/dashboard/node_modules/gulp/node_modules/orchestrator/index.js:275:23
    at finish (/home/ec2-user/dashboard/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
    at cb (/home/ec2-user/dashboard/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:29:3)
    at finish (/home/ec2-user/dashboard/node_modules/run-sequence/index.js:48:5)
    at Gulp.onError (/home/ec2-user/dashboard/node_modules/run-sequence/index.js:55:4)
    at Gulp.emit (events.js:117:20)
```

It appears you are checking to see if `accessKeyId` is a member of opts: 
https://github.com/pgherveou/gulp-awspublish/blob/e7f8400448cf38ed016659a1100717e99e987d83/lib/index.js#L38-L43

The on an EC2 instance with an IAM Instance Profile, the AWS access id and secret key will not be set in the environment, and the AWS SDK will search for credentials in the instance-metadata. (It will also look in the environment, as well.) Since you have switched to use aws-sdk (vs knox), I don't believe this check is necessary.

If the credentials are incorrectly set by the user, why not let the error bubble up from the AWS SDK? (ie: Remove lines 38 thru 43 in `function getCredentials()`)

**Comment by pgherveou** [comment]
Yes agree letting the error bubble up make sense to me
 @mshick wrote the aws integration, what do you think?

**Comment by pgherveou** [comment]
@vrivellino if things works for you without https://github.com/pgherveou/gulp-awspublish/blob/e7f8400448cf38ed016659a1100717e99e987d83/lib/index.js#L38-L43
can you make a pull request?

**Comment by pgherveou** [comment]
@vrivellino did you try simply to make your code go through this condition https://github.com/pgherveou/gulp-awspublish/blob/e7f8400448cf38ed016659a1100717e99e987d83/lib/index.js#L71

**Comment by vrivellino** [comment]
@pgherveou I believe I did. I was just using some console.log() statements, but it appears the first time through the call to GetCredentials(), it fell all the way through down to line 75: `return getCredentials(new AWS.SharedIniFileCredentials(opts));` And it bailed when it made the second call to GetCredentials().

I have confirmed that deleting lines 38-43 corrects this problem both on an EC2 instance with an Instance Profile, as well as my local system (where I have AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY set in my environment).  Also, when those environment variables are not set, the AWS SDK blows everything up, as expected.
